src/mito/src/table.rs

```diff
@@ -70,7 +70,7 @@ pub struct MitoTable<R: Region> {
     manifest: TableManifest,
     // guarded by `self.alter_lock`
     table_info: ArcSwap<TableInfo>,
-    regions: HashMap<RegionNumber, R>,
+    regions: Arc<RwLock<HashMap<RegionNumber, R>>>,
```

Comment:
Performance optimization suggestion: can you use "[arc-swap](https://crates.io/crates/arc-swap)"? So you can make `regions` of type `ArcSwap<HashMap<RegionNumber, R>>`. Because the `regions` are seldom updated, it's faster than acquiring the read lock everytime.

Then introduce some kind of "copy on write" mechanism, when `regions` needs to be updated, clone the whole regions `HashMap`(instead of `Arc::clone`), update the new hashmap and set it back here. 

There might be some locks required, but they can be put inside each region's codes.

Besides, this can get rid of the propagation of `await`.